### PR TITLE
Add element matching for multiple values on text/href

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- Previously, if you try to use an element filter on events with multiple values, the query would fail because the filter function could only handle single values
- this implements support for multiple values and updates tests to reflect changes
*If this affects the frontend, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: ee/clickhouse/models/action.py [hunk 1]
```diff
@@ -116,16 +116,22 @@ def filter_element(filters: Dict, prepend: str = "") -> Tuple[List[str], Dict]:
         params["{}tag_name_regex".format(prepend)] = r"(^|;){}(\.|$|;|:)".format(filters["tag_name"])
         conditions.append("match(elements_chain, %({}tag_name_regex)s)".format(prepend))
```
Comment (macobo) [inline ee/clickhouse/models/action.py:117]: WDYT about move this function together with the rest of the filters logic files over at ee/clickhouse/models/property.py

File: posthog/models/event.py [hunk 2]
```diff
@@ -146,10 +147,18 @@ def filter_by_element(self, filters: Dict, team_id: int):
...
-        if not filter:
+            vals = filters.get(key)
+            if vals:
+                if isinstance(vals, str):
+                    filter["element__{}".format(key)] = filters[key]
+                elif isinstance(vals, list):
+                    _filter_conditions = Q()
+                    for val in vals:
+                        _filter_conditions |= Q(**{"element__{}".format(key): val})
+                    groups = groups.filter(_filter_conditions)
```
Comment (macobo) [inline posthog/models/event.py:158]: This would result in multiple queries being made I _think_?

File: posthog/models/event.py [hunk 3]
```diff
@@ -146,10 +147,18 @@ def filter_by_element(self, filters: Dict, team_id: int):
             filter = {}
 
         for key in ["tag_name", "text", "href"]:
-            if filters.get(key):
-                filter["element__{}".format(key)] = filters[key]
-
-        if not filter:
+            vals = filters.get(key)
```
Comment (macobo) [inline posthog/models/event.py:150]: I think we can simplify this code. The `did_filter`, etc logic is kind of awkward.

WDYT about this:

```python
filters = Q()
for key in ["tag_name", "text", "href"]:
    values = filters.get(key, [])
    values = values if isinstance(values, list) else [values]
    if len(values) == 0: continue

    condition = Q()
    for searched_value in values:
        condition |= Q(**{"element__{}".format(key): searched_value}
    filters &= condition

if not filters: return {}
groups = groups.filter(filters)
```

It does have some slight difference of behavior if filters[key] is None - I think that's not possible?

**Review by macobo — COMMENTED** [review]
Might be worth adding a FE test for this flow as well - since it's separate from rest of filtering logic, it's easy to break this as our filtering evolves.

**Comment by EDsCODE** [comment]
Woah I somehow pushed as @fuziontech  on my local env?

File: posthog/models/event.py (hunk 3 above)
Comment (EDsCODE) [reply posthog/models/event.py]: Ah this `None` issue was the problem! I tried something similar above last night but it was breaking a lot of tests so i left the smelly code in. The action steps that are passed in have None for ["tag_name", "text", "href"] so it does need to be handled. Thanks for the pointer

File: ee/clickhouse/models/action.py (hunk 1 above)
Comment (EDsCODE) [reply ee/clickhouse/models/action.py]: Done! I reflexively tried to go to property tests too to find the tests for this function

Comment (EDsCODE) [reply ee/clickhouse/models/action.py]: It does however cause a circular import so we'll have to import within a function to resolve it

Comment (macobo) [reply ee/clickhouse/models/action.py]: I think that's fine :)

**Review by macobo — APPROVED** [review]

**Comment by EDsCODE** [comment]
Going to get this in before the code freeze without frontend tests

**Pull request merged**